iofog-agent-daemon/src/test/java/org/eclipse/iofog/command_line/CommandLineActionTest.java

```diff
@@ -0,0 +1,408 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import javax.json.Json;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.Mockito.*;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+
+
+/**
+ * @author nehanaithani
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({CommandLineAction.class, StatusReporter.class, FieldAgent.class, Configuration.class,
+        Orchestrator.class})
+public class CommandLineActionTest {
+    private CommandLineAction commandLineAction;
+    private StatusReporter statusReporter;
+    private FieldAgent fieldAgent;
+    private List stop;
+    private HashMap<String, String> result;
+
+    @Before
+    public void setUp() throws Exception {
+        commandLineAction = mock(CommandLineAction.class);
+        statusReporter = mock(StatusReporter.class);
+        fieldAgent = mock(FieldAgent.class);
+        mockStatic(FieldAgent.class);
+        mockStatic(Orchestrator.class);
+        mockStatic(Configuration.class);
+
+        PowerMockito.when(FieldAgent.getInstance()).thenReturn(fieldAgent);
+
+
+        PowerMockito.when(fieldAgent.provision("dummy")).thenReturn(Json.createObjectBuilder()
+                .add("status", "success")
+                .add("errorMessage", "")
+                .add("uuid", "uuid")
+                .build());
+        PowerMockito.when(fieldAgent.provision("anotherkey")).thenReturn(Json.createObjectBuilder()
+                .add("status", "success")
+                .add("errorMessage", "Key not valid")
+                .build());
+        PowerMockito.when(fieldAgent.provision("prod")).thenReturn(null);
+
+        // PowerMockito.when(Configuration.setConfig(anyMap(), anyBoolean())).thenReturn(null);
+        stop = new ArrayList(Collections.singleton("stop"));
+
+        result = new HashMap<>();
+        result.put("ll", "info");
+    }
+
+
+    @After
+    public void tearDown() throws Exception {
+        stop = null;
+        result = null;
+        reset(statusReporter);
+        reset(fieldAgent);
+    }
+
+    /**
+     * Get Keys method
+     */
+    @Test
+    public void getKeys() {
+
+        assertFalse(CommandLineAction.HELP_ACTION.getKeys().isEmpty());
+        assertTrue(isEqual(stop, CommandLineAction.getActionByKey("stop").getKeys()));
+    }
+
+    /**
+     * help command with success
+     */
+    @Test
+    public void helpActionPerform() {
+        String[] helpArgs = {"help", "-h", "-?"};
+        assertEquals(helpContent, CommandLineAction.getActionByKey(helpArgs[0]).perform(helpArgs));
```

Comment:
Please add these test cases as well:
```
assertEquals(helpContent, CommandLineAction.getActionByKey(helpArgs[1]).perform(helpArgs))
assertEquals(helpContent, CommandLineAction.getActionByKey(helpArgs[2]).perform(helpArgs))
assertEquals(helpContent, CommandLineAction.getActionByKey(helpArgs[3]).perform(helpArgs))
```
Same thing for `version`

Reply:
Updated